parley/src/resolve/mod.rs

```diff
@@ -499,7 +512,15 @@ impl<B: Brush> ResolvedStyle<B> {
             brush: self.brush.clone(),
             underline: self.underline.as_layout_decoration(&self.brush),
             strikethrough: self.strikethrough.as_layout_decoration(&self.brush),
-            line_height: self.line_height * self.font_size,
+            line_height: match self.line_height {
+                crate::LineHeight::MetricsRelative(value) => {
+                    crate::LayoutLineHeight::MetricsRelative(value)
+                }
+                crate::LineHeight::FontSizeRelative(value) => {
+                    crate::LayoutLineHeight::Absolute(value * self.font_size)
+                }
+                crate::LineHeight::Absolute(value) => crate::LayoutLineHeight::Absolute(value),
+            },
```

Comment:
Same as `nearly_eq` above (considering `use crate::LineHeight` + make this a method).